packages/eslint-plugin/tests/rules/no-unnecessary-type-parameters.test.ts

```diff
@@ -211,6 +211,7 @@ ruleTester.run('no-unnecessary-type-parameters', rule, {
     `,
     'declare function get(): void;',
     'declare function get<T>(param: T[]): T;',
+    'declare function factory<T>(): T[];',
```

Comment:
(If we don't have it already) Add a test like:

```ts
declare function returnsTuple<T>(): [T]
```

This should be invalid, unlike returning `T[]` which we're leaving valid.

Reply:
Added :D